tests/checkup_test.go

```diff
@@ -0,0 +1,237 @@
+	var kiagnoseConfigMapAccessRole *rbacv1.Role
+	var kiagnoseConfigMapAccessRoleBinding *rbacv1.RoleBinding
+	var configMap *corev1.ConfigMap
+	var checkupJob *batchv1.Job
+
+	BeforeEach(func() {
+		var err error
+
+		checkupServiceAccount = newServiceAccount()
+		_, err = client.CoreV1().ServiceAccounts(testNamespace).Create(context.Background(), checkupServiceAccount, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		kiagnoseConfigMapAccessRole = newKiagnoseConfigMapAccessRole()
+		_, err = client.RbacV1().Roles(testNamespace).Create(context.Background(), kiagnoseConfigMapAccessRole, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		kiagnoseConfigMapAccessRoleBinding = newKiagnoseConfigMapAccessRoleBinding()
+		_, err = client.RbacV1().RoleBindings(testNamespace).Create(
+			context.Background(),
+			kiagnoseConfigMapAccessRoleBinding,
+			metav1.CreateOptions{},
+		)
+		Expect(err).NotTo(HaveOccurred())
+
+		configMap = newConfigMap()
+		_, err = client.CoreV1().ConfigMaps(testNamespace).Create(context.Background(), configMap, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		checkupJob = newCheckupJob()
+		_, err = client.BatchV1().Jobs(testNamespace).Create(context.Background(), checkupJob, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		DeferCleanup(func() {
+			backgroundPropagationPolicy := metav1.DeletePropagationBackground
+			err = client.BatchV1().Jobs(testNamespace).Delete(
+				context.Background(),
+				testCheckupJobName,
+				metav1.DeleteOptions{PropagationPolicy: &backgroundPropagationPolicy},
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = client.CoreV1().ConfigMaps(testNamespace).Delete(context.Background(), testConfigMapName, metav1.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			err = client.RbacV1().RoleBindings(testNamespace).Delete(
+				context.Background(),
+				testKiagnoseConfigMapAccessRoleBindingName,
+				metav1.DeleteOptions{},
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = client.RbacV1().Roles(testNamespace).Delete(context.Background(), testKiagnoseConfigMapAccessRoleName, metav1.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			err = client.CoreV1().ServiceAccounts(testNamespace).Delete(context.Background(), testServiceAccountName, metav1.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
+	It("should complete successfully", func() {
+		Eventually(getJobConditions, 5*time.Minute, 5*time.Second).Should(
+			ContainElement(MatchFields(IgnoreExtras, Fields{
+				"Type":   Equal(batchv1.JobComplete),
+				"Status": Equal(corev1.ConditionTrue),
+			})))
+
+		configMap, err := client.CoreV1().ConfigMaps(testNamespace).Get(context.Background(), testConfigMapName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(configMap.Data).NotTo(BeNil())
+		Expect(configMap.Data["status.succeeded"]).To(Equal("true"))
+		Expect(configMap.Data["status.failureReason"]).To(BeEmpty())
```

Comment:
It will be hard to debug a failure here.
We will not see the whole picture of what this config-map contains.

If it is not possible to assert the whole object, then at least report its whole content on failure (through the assert description possibly).

Reply:
The matchers have a "description" option.
Example: `Expect(prepare.executed).To(BeTrue(), "the prepare step should execute")`

In that text you can add more info, like the content of all the configmap.

Reply:
This line prints the CM data on failure:
```go
Expect(configMap.Data["status.succeeded"]).To(Equal("true"), fmt.Sprint(configMap.Data))
```
WDYT?

Reply:
Yes, something like that. Maybe add “should succeed” for some context. Up to you.

Reply:
Done.